### DISCUSSION TITLE
Sign in with Microsoft in the JS SDK layer on iOS / Android

### DISCUSSION THREAD

**Discussion by RaduMirceaAndrei** [capawesome-team/capacitor-firebase#392] [Q&A] [answered]
Hello,

I'm not sure if this is a discussion worthy issue, or a bug.
I'm having difficulties trying to sign in with Microsoft as the provider, in the JS SDK layer, on both iOS and Android.
After successfully signing in to Firebase, i'm grabbing the token Microsoft sends back and trying to authenticate with that in the JS SDK. This is what i am doing:
```js
FirebaseAuthentication.signInWithMicrosoft().then(async signInResult => {
        const provider = new OAuthProvider('microsoft.com');
        const credential = provider.credential({
            idToken:signInResult.credential?.idToken
        });
        const auth = getAuth();
        signInWithCredential(auth, credential).then(async (userCredential) => {
            console.log('login successfully');
        }).catch(e => {
            console.log(e);
        });
    });
```
The problem is Firebase is telling me the idToken that i just provided is invalid. In a browser environment, without the signInWithCredential part, i am logged in to the JS, so i would expect that it would work the same on mobile, since there's no native popup or sign in mechanism for Microsoft (it does not, and i am not signed in, that's why i've attempted the signInWithCredential).
The difference is, as far as i can tell, that in a mobile environment, the plugin opens a custom browser tab on top of my app and grabs credentials from it, without signing in to the JS SDK - credentials that cannot be used to signInWithCredentials, like for Google or Apple (which work).

So to sum it up, i don't know how to sign in to Firebase in the JS SDK layer, on mobile. Or if it is at all possible.
I've looked trough the documentation, however i didn't find anything specifically related to this.

I would appreciate any help, guidance, clarification or link to some documentation regarding this.

Thank you very much for your time.

**Comment by robingenz** [comment, marked as answer]
Please take a look at https://github.com/capawesome-team/capacitor-firebase/discussions/216#discussioncomment-3803525

**Reply by RaduMirceaAndrei** [reply]
Got it. I'll mark this as answered. Thank you for your time.
